scripts/schema_bump_dry_run_ontologies/ontology_bump_dry_run.py

```diff
@@ -63,37 +55,30 @@ def map_deprecated_terms(
                 for ontology_term_id in ontology_term_ids:
                     if ontology_term_id in non_deprecated_term_cache:
                         continue
-                    else:
-                        # all_ontologies is indexed by ontology prefix and term ID without suffixes,
-                        # so we must parse + build the search term
-                        ontology_id_parts = ontology_term_id.split(" ")[0].split(":")
-                        term_prefix = ontology_id_parts[0]
-                        ontology_index_id = f"{term_prefix}:{ontology_id_parts[1]}"
-                        ontology = onto_map[term_prefix][ontology_index_id]
-
-                    if ontology["deprecated"]:
+                    if onto_parser.is_term_deprecated(ontology_term_id):
                         if ontology_term_id in curator_report_entry_map[collection_id]:
                             curator_report_entry_map[collection_id][ontology_term_id]["dataset_ct"] += 1
                         else:
                             entry = dict()
                             entry["needs_alert"] = False
                             entry["dataset_ct"] = 1  # type: ignore
-                            if "term_tracker" in ontology:
-                                entry["term_tracker"] = ontology["term_tracker"]
-                            if "comments" in ontology:
-                                entry["comments"] = ontology["comments"]
-                            if "replaced_by" in ontology:
-                                entry["replaced_by"] = ontology["replaced_by"]
-                                replacement_term_ontology = ontology["replaced_by"].split(":")[0]
-                                if replacement_term_ontology != term_prefix:
+                            ontology_metadata = onto_parser.get_term_metadata(ontology_term_id)
+                            replacement_term = onto_parser.get_term_replacement(ontology_term_id)
```

Comment:
nit: move this right above `if replacement_term:` since that is where it's used.